crates/sui-types/src/message_envelope.rs

```diff
@@ -5,14 +5,41 @@ use crate::base_types::AuthorityName;
 use crate::committee::{Committee, EpochId};
 use crate::crypto::{
     AuthorityQuorumSignInfo, AuthoritySignInfo, AuthoritySignInfoTrait, AuthoritySignature,
-    EmptySignInfo, Signable,
+    AuthorityStrongQuorumSignInfo, EmptySignInfo, Signable,
 };
 use crate::error::SuiResult;
+use crate::messages_checkpoint::CheckpointSequenceNumber;
 use once_cell::sync::OnceCell;
 use serde::{de::DeserializeOwned, Deserialize, Serialize};
 use std::fmt::{Debug, Display, Formatter};
 use std::ops::Deref;
 
+/// IndirectValidity is a placeholder for signatures, which indicates that the wrapped message has
+/// been proven valid through indirect means, typically inclusion in a certified checkpoint or
+/// via f+1 votes that the message is correct.
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct IndirectValidity(Validity);
```

Comment:
These don't seem to belong to message_envelope.rs. Maybe either in messages.rs or a separate file.

Reply:
moved to new file